keps/sig-cli/1441-kubectl-debug/README.md

```diff
@@ -300,6 +307,124 @@ Examples:
+set `SYS_ADMIN` while debugging a node, a pod by ephemeral container, or a pod by copy.
+`kubectl debug` is intended to guide the user through a debugging scenario, and
+requiring the user to specify a series of flags on the command line is a poor experience.
+
+Instead, we'll introduce "Debugging profiles" which are configurable via a single command
+line flag, `--profile`.  A user may then use, for example, `--profile=netadmin` when
+debugging a node to create a pod with the `NET_ADMIN` capaibility.
+
+The available profiles will be:
+
+| Profile      | Description                                                     |
+| ------------ | --------------------------------------------------------------- |
+| debug        | A reasonable set of defaults tailored for each debuging journey |
+| baseline     | Compatible with baseline [Pod Security Standard]                |
+| restricted   | Compatible with restricted [Pod Security Standard]              |
+| auto         | Automatically choose between debug, baseline, and restricted    |
+| sysadmin     | System Administrator (root) privileges                          |
+| netadmin     | Network Administrator privileges.                               |
+| legacy       | Backwards compatibility with 1.22 behavior                      |
+
+Debugging profiles are intended to work seamlessly with the [Pod Security Standard]
+enforced by the [PodSecurity] admission controller. The baseline and restricted
+profiles will generate configuration compatible with the corresponding security
+level.
+
+It might be possible to support user-configurable profiles, but it's not a goal of
+this KEP, and we have no plans to implement it.
+
+[Pod Security Standards]: https://kubernetes.io/docs/concepts/security/pod-security-standards/
+[PodSecurity]: http://kep.k8s.io/2579
+
+#### Default Profile and Automation Selection
+
+The profile named "debug" will generate the best configuration for a particular
+debugging journey, for example including `SYS_PTRACE` for ephemeral containers
+so that a debugging container may signal and attach to other processes.
+
+The profile named "auto" will choose between "debug", "baseline", and "restricted"
+by looking `pod-security.kubernetes.io/enforce` annotation on the namespace and
+choosing the most permission of "debug", "baseline", and "restricted" that will
+be allowed by default. In this way it will choose a default that's always
+compatible with the current security policy.
+
+When not `--profile` is not specified it will default to "legacy" to be backwards
+compatible with the current behavior and print a warning that the behavior will
+change starting with release 1.25. Starting with 1.25 the default will be "auto".
+
+Even with automatic selection it's still desirable to be able to specify "debug",
+"baseline", and "restricted" explicitly. In the future there will be an optional
+"break glass" mechanism to bypass the current security policy for debugging.
```

Comment:
locating this in this doc is a bit weird... it's probably not on `kubectl debug` to define breakglass mechanisms for security policies on the cluster, and `kubectl debug` can't guarantee it can always work no matter what policies are in place

the current mechanisms for relaxing PodSecurity admission are:
* temporarily removing or relaxing the enforce label on the namespace
* creating the ephemeral container/pod as a user that is configured as exempt from PodSecurity enforcement (though `kubectl debug` won't have visibility to know what user(s) that could be)

Reply:
I expected there to be a standard way to configure break-glass that `kubectl debug` would support, especially since it's planned to change the default to baseline in the future.

It'd be easy enough to have `kubectl debug` support both the mechanisms listed.

Reply:
> especially since it's planned to change the default to baseline in the future.

individual deployments/clusters could choose to change that default, but there is no plan to change that default at the kube-apiserver level...

Reply:
Maybe move that to improvements section?